src/General-Rules/ReCodeCruftLeftInMethodsRule.class.st

```diff
@@ -121,3 +121,12 @@ ReCodeCruftLeftInMethodsRule >> rationale [
 ReCodeCruftLeftInMethodsRule >> severity [
 	^ #error
 ]
+
+{ #category : 'testing' }
+ReCodeCruftLeftInMethodsRule >> shouldCheckMethod: aMethod [
+
+	(aMethod hasPragmaNamed: #debuggerCompleteToSender)
+	| (aMethod hasPragmaNamed: #haltOrBreakpointForTesting) ifTrue: [
+		^ false ].
```

Comment:
```suggestion
	(#(#debuggerCompleteToSender #haltOrBreakpointForTesting) anySatifsy: [ :selector |  aMethod hasPragmaNamed: selector ]) ifTrue: [ ^ false ].
```

Most of the time in Pharo we should use #or: instead of #|
#or: is usually better. In the past we still used #| for speed reasons but now there is an optimization on #or: making it as fast I think.

Also we can avoid totally the or: like in the code I proposed. This would make it easier in the future to add more pragmas.

Reply:
For |, both sides have to be evaluated. For or:, if the left hand is true, it does not need to evaluate the argument, for | both sides are evaluated. This makes or: in many cases faster. In addition, or: is optimized to jump (like ifTrue:), while | is a message send.